crates/jrsonnet-evaluator/src/lib.rs

```diff
@@ -221,11 +226,11 @@ impl EvaluationState {
 		Ok(self.data().str_files.get(&path).cloned().unwrap())
 	}
 
-	fn evaluate_loaded_file_raw(&self, name: &PathBuf) -> Result<Val> {
+	fn evaluate_loaded_file_raw(&self, name: &Path) -> Result<Val> {
 		let expr: LocExpr = {
 			let ro_map = &self.data().files;
 			let value = ro_map
-				.get(name)
+				.get(&name.to_path_buf())
```

Comment:
Better use Path as map keys, or receive PathBuf as this function args

Reply:
From what I understand of Path/PathBuf, if you need to store the data use `PathBuf`, otherwise use `Path`. Since the EvaluationData holds the path for the file - although, inside an `Rc` - I think we should stick to `PathBuf` instead of `Path`.

Using Path as map keys causes too many changes and seem to be wrong. I would use `PathBuf` as the argument, but this moves the transformation, which appears to be your worry, to somewhere else. Also, it goes against the suggestion, from the clippy warning, of using slices instead of "owned types".

I'm new to Rust, and this is my first contribution. If what I said is wrong, please let me know. But from what I understand, the course of action should be to disable this warning instead, as the two latter options are as optimal as disabling it.

Another option would be to leave as it's, but I would like to understand why this got your attention. Is the new conversion/clone a performance concern?

Reply:
I will look at this PR later, and probably accept it anyway, so do not close it yet, and thanks for contribution!